src/type_seed.jl

```diff
@@ -0,0 +1,76 @@
+"""
+    type_seed(x)
+
+Computes a value to use as a seed for computing the hash value of a type.
+"""
+function type_seed end
+
+function type_seed(m::Module, h::UInt)
+    if m === Main || m === Base || m === Core
+        return h
+    else
+        mp = parentmodule(m)
+        if mp === m
+            return h
+        else
+            return hash(nameof(m), type_seed(mp, h))
+        end
+    end
+end
+
+function type_seed(t::UnionAll, h::UInt)
+    h = hash(h, 0xbd5f3e4941dba79d)
+    h = type_seed(t.var, h)
+    h = type_seed(t.body, h)
+    h
```

Comment:
```suggestion
    return h
```
for stylistic constistency